### ISSUE TITLE
[no-unused-vars] Incorrect line number on import statement in Vue + TypeScript

### ISSUE THREAD

**Issue by nokazn** [typescript-eslint/typescript-eslint#2735]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Reproduction repository for this issue is [here](https://github.com/nokazn/typescript-eslint-vue-repro).

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```Vue
<template>
  <div>
    <!-- actually "no-unused-vars" error is reported at line 4 -->
    <p>foo</p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
// the error should be reported at line 11
import Button from './Button.vue';

export default Vue.extend({});
</script>
```

**Expected Result**

An error is reported like `11:8  error  'Button' is defined but never used  @typescript-eslint/no-unused-vars` in the above example.

**Actual Result**

An error `4:8  error  'Button' is defined but never used  @typescript-eslint/no-unused-vars` is reported.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.6.1` |
| `@typescript-eslint/parser`        | `4.6.1` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.12.1` |
| `node`                             | `12.18.0` |

**Comment by bradzacher** [comment]
We don't really support vue here.

I would suggest raising an issue in `vue-eslint-parser`, which does some magic on top of our parser and scope analysis. It's a big black box that is outside my wheelhouse.

https://github.com/vuejs/vue-eslint-parser

They would have a better chance of being able to investigate and find the problem.
